src/inmanta/util.py

```diff
@@ -404,21 +404,27 @@ def datetime_iso_format(timestamp: datetime.datetime, *, naive_utc: bool = False
     :param tz_aware: Whether to return timezone aware timestamps or naive, implicit UTC timestamp.
     """
 
-    if tz_aware:
-        if timestamp.tzinfo:
-            return timestamp.isoformat(timespec="microseconds")
-        if naive_utc:
-            return timestamp.replace(tzinfo=datetime.timezone.utc).isoformat(timespec="microseconds")
-        return timestamp.astimezone().isoformat(timespec="microseconds")
+    def convert_timestamp(timestamp: datetime.datetime, naive_utc: bool, tz_aware: bool) -> datetime.datetime:
```

Comment:
I would keep it simple and drop the parameters since they're exactly the same as the ones passed to the surrounding method.
```suggestion
    def convert_timestamp() -> datetime.datetime:
```